### PR TITLE
DB-9173 Select on SYS.SYSCOLUMNS fails when there is a table with column default specified.

### PR DESCRIPTION
**Pull request by ipraznik-splice**

### CODE REVIEW

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java [hunk 1]
```diff
@@ -360,16 +360,25 @@ public void updateColumnViewInSysIBM(TransactionController tc) throws StandardEx
...
+            SpliceLogUtils.info(LOG, String.format("The view %s in %s has been updated with default column!", tableName, schemaDescriptor.getSchemaName()));
         }
     }
 
+
+    public void updateColumnViewInSysIBM(TransactionController tc) throws StandardException {
+        updateColumnViewInSys(tc, "SYSCOLUMNS", 1, sysIBMSchemaDesc, SYSCOLUMNSRowFactory.SYSCOLUMNS_VIEW_IN_SYSIBM);
+    }
+
+    public void updateColumnViewInSysVM(TransactionController tc) throws StandardException {
```
[the remark this answers is not in the document]
Comment (ipraznik-splice) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java]: fixed

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/UpgradeScriptForAddDefaultToColumnViewInSYSVM.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @Override
+    protected void upgradeSystemTables() throws StandardException {
+        sdd.updateColumnViewInSysVM(tc);
+
+        // we need to re-generate the metadataSPS if it is a change of definition of the table/column views in sysVM
+        sdd.updateMetadataSPSes(tc);
+
+        SpliceLogUtils.info(LOG, "Catalog upgraded: updated syscolumns view in SYSVM schema");
```
[the remark this answers is not in the document]
Comment (ipraznik-splice) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/UpgradeScriptForAddDefaultToColumnViewInSYSVM.java]: fixed

[the remark this answers is not in the document]
Comment (ipraznik-splice) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/UpgradeScriptForAddDefaultToColumnViewInSYSVM.java]: the upgrade has been tested, as you described

[the remark this answers is not in the document]
Comment (ipraznik-splice) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/UpgradeScriptForAddDefaultToColumnViewInSYSVM.java]: jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/SpliceCatalogUpgradeScripts.java [hunk 3]
```diff
@@ -83,6 +83,7 @@ else if(v1==v2)
         scripts.put(new Splice_DD_Version(sdd,3,1,0, 1953), new UpgradeScriptForRemoveUnusedIndexInSYSFILESTable(sdd,tc));
         scripts.put(new Splice_DD_Version(sdd,3,1,0, 1954), new UpgradeScriptToInvalidateStoredStatement(sdd,tc));
         scripts.put(new Splice_DD_Version(sdd,3,1,0, 1959), new UpgradeScriptForTriggerMultipleStatements(sdd,tc));
+        scripts.put(new Splice_DD_Version(sdd,3,1,0, 1961), new UpgradeScriptForAddDefaultToColumnViewInSYSVW(sdd,tc));
```
Comment (arnaud-splice) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/SpliceCatalogUpgradeScripts.java:86]: If we target this week's build, you will need to set this to 1962

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/SpliceCatalogUpgradeScripts.java (hunk 3 above)
Comment (ipraznik-splice) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/SpliceCatalogUpgradeScripts.java]: The version has been updated

**Pull request merged**